src/basic/process-util.h

```diff
@@ -148,6 +148,8 @@ static inline bool sched_priority_is_valid(int i) {
         return i >= 0 && i <= sched_get_priority_max(SCHED_RR);
 }
 
+#define PID_INVALID ((pid_t) -1)
```

Comment:
not sure how i feel about this one, given that we usually set 0 as invalid PID, i.e. TAKE_PID() for example

Reply:
what about this idea, maybe turn this around and introduce a special value `PID_AUTOMATIC` or so as `((pid_t) INT_MIN)` or so, and use that as marker for "determine the pid automatically".

This would be very literal then and we could continue to store regular invalid pids as 0 everywhere. 

This would then mean the json dispatch would map json null → to 0, but json "0" → PID_AUTOMATIC. (which is slightly weird, as we mangle zeroes and nulls a bit here and swap their meaning, but I think there's a good reason for that.

Reply:
What do you suggest? If I do PID_INVALID=0 then a request like `io.systemd.Machine.List '{"pid": null}'` would be indistinguishable from `io.systemd.Machine.List '{"pid": 0}' unless I miss something.

Maybe I can mark pid fields as non-nullable to work around. Need to check how it will work.

Reply:
i am suggesting this:

```c
#define PID_AUTOMATIC ((pid_t) INT_MIN) /* special value indicating "acquire pid from connection peer */
#define PID_INVALID ((pid_t) 0) /* default value for "invalid pid" */
```

and the dispatcher would parse json value 0 into PID_AUTOMATIC, and json value null into PID_INVALID.

Hence all cases would be nicely distinguishable